### DISCUSSION TITLE
SUMO vehicles collision

### DISCUSSION THREAD

**Discussion by nikolatopic131** [carla-simulator/carla#7860] [Q&A]
Hello everyone. I created two maps in Roadrunner to use for CARLA-SUMO co-simulations. I've netconverted the opendrive to .net.xml to get the SUMO maps. However, when I run my co-simulation using run_synchronization.py, my SUMO vehicles can pass through the objects like fences etc. Does anyone know if it is possible to configure it so that they can't pass through objects, but collide instead?

**Comment by arscode** [comment]
Hi, do you know how to run co-simulation any specific scenario say leader-follower or driving around a roundabout? Also I'm trying to run my own traci script with co-simulation, any idea how to do it?

**Reply by nikolatopic131** [reply]
Hello. I'm not sure if I understand your question. Do you mean how to run a co-simulation with a custom scenario, as in, not one of the maps already given by CARLA? If so, for the map part, I've created a map in Roadrunner and then ingested it into CARLA, as per the directions given in the documentation. As for the traci code part, do you mean you have a SUMO simulation which is being controlled by the traci api, and you want to run it as a sumo-carla co-simulation? If so, you can copy your sumo simulation code into the "synchronization loop" function part of the run_synchronization.py script provided by CARLA, and then run the co-simulation with it. Doing so will run a carla-sumo co-simulation with the sumo simulation being controlled by your traci code (that's how I did it anyway, there are probably more efficient ways but I'm not very good in development).

Edit; also for running a co-simulation you need a sumocfg file and your sumo and carla maps need to be matching with matching projections otherwise the vehicles in carla will drive over empty space. You run the co-simulation by running the run_synchronization.py script through a terminal/command prompt and passing the path to your .sumocfg file as an arguement.

**Reply by arscode** [reply]
Hi, Thanks a lot for your detailed reply. I'm not trying to use new maps, I'm mostly working with CARLA built-in maps for now. Thanks you've cleared my confusion where and ow to push my TraCI code as I was stuck due to the server-client architecture of both of the software. I guess I have to push my code in the while-loop inside the "synchronizaiton loop". I also built some carla code as well , is that can also be used in a same way?

Thanks again for your help..!!

**Reply by nikolatopic131** [reply]
Sorry for the late reply. What kind of carla code? If it is for spawning vehicles then yes I believe it has to go into the same loop (hadn't yet reached the step of adding carla agents into my simulation, having some issues with the map, so I'm not sure how it works), if it is about something else then perhaps it has to go elsewhere.

**Reply by arscode** [reply]
yeah my code spawning and navigating the vehicle using a waypoint. In that case, the sumo detecting the CARLA vehicle as 'carla0'  but I guess it maps the carla vehicle into sumo simulation using vtype and *.rou.xml when(the departure time in route file) that vtype matches. before that the CARLA vehicle not present in the sumo simulation. also the CARLA vehicle remain int he sumo simulation as long as it lane id list in route file not exhausted, whereas in the CARLA simulation I can let it run infinitely.

**Reply by arscode** [reply]
> Sorry for the late reply. What kind of carla code? If it is for spawning vehicles then yes I believe it has to go into the same loop (hadn't yet reached the step of adding carla agents into my simulation, having some issues with the map, so I'm not sure how it works), if it is about something else then perhaps it has to go elsewhere.

have you used fbx and xodr file as well for your custom designed map? It seems the CARLA not detecting the object so it has no effect in the co-simulation as well. have you checked the autogenerate collisions feature of your objects in the custom map?

you can check this paper and their github. they have created a roundabout map using roadrunner:
https://hal.science/hal-04141974/file/publi-7142.pdf
